### PR TITLE
Fix #1091

### LINKED ISSUE #1091: Difference of Newtype behaviour in pattern matching between RC16 and RC17
Hi,

I tried to update zio-prelude in our project, but some tests fail because the behaviour of newtypes in pattern matching has changed between RC16 and RC17.

This prints `"toto"` with RC16 but prints `"tata"` with RC17:
```scala
import zio.prelude._
import zio.prelude.Assertion._

object Id extends Subtype[Int] {
  override def assertion = assert(greaterThanOrEqualTo(0))
}
type Id = Id.Type

Id(1) match {
  case Id(1) => println("toto")
  case _     => println("tata")
}
```

I'm not sure which behaviour is correct. I'd say that the one from RC16 is the correct one

RC16: https://scastie.scala-lang.org/cvjxJq3JQ6Czy5Vg0jqE5g
RC17: https://scastie.scala-lang.org/OrlZvhDrSDmfLqzgmLgCcQ

### PR DESCRIPTION
**Pull request by guizmaii**
Fix #1091

Related PR:
- https://github.com/zio/zio-prelude/pull/1054

### CODE REVIEW

**Comment by adamgfraser** [comment]
@mschuwalow Can you take a look?

File: core/shared/src/main/scala-2/zio/prelude/Newtype.scala [hunk 1]
```diff
@@ -284,7 +284,13 @@ abstract class Newtype[A] extends NewtypeVersionSpecific {
    */
   def wrapAll[F[_]](value: F[A]): F[Type] = macro zio.prelude.Macros.wrapAll_impl[F, A, Type]
 
-  implicit def classTag(implicit underlying: ClassTag[A]): ClassTag[Type] = Newtype.classTag
+  /**
```
Comment (sideeffffect) [inline core/shared/src/main/scala-2/zio/prelude/Newtype.scala]: Shouldn't this rather be a regular comment with `/*` instead of `/**`?

Comment (guizmaii) [reply core/shared/src/main/scala-2/zio/prelude/Newtype.scala]: Fixed :)

**Comment by guizmaii** [comment]
Closed in favour of https://github.com/zio/zio-prelude/pull/1094

**Pull request closed without merging**